umbra-js/src/classes/Umbra.ts

```diff
@@ -417,17 +418,23 @@ export class Umbra {
 
   /**
    * @notice Fetches Umbra event logs starting from the block user registered their stealth keys in using
-   * Goldsky, if available, falling back to RPC if not
+   * a subgraph, if available, falling back to RPC if not
+   * @param possibleRegisteredBlockNumber Block number when user registered their stealth keys (if known)
+   * @param Signer Signer with provider to use for fetching the block number (if not known) from the StealthKeyRegistry contract
+   * @param address Address of the user for fetching the block number (if not known) from the subgraph or StealthKeyRegistry contract
    * @param overrides Override the start and end block used for scanning;
    * @returns A list of Announcement events supplemented with additional metadata, such as the sender, block,
    * timestamp, and txhash
+   * @dev If the registered block number is not known, it will be fetched from the subgraph or the StealthKeyRegistry contract
    */
   async *fetchSomeAnnouncements(
+    possibleRegisteredBlockNumber: number | undefined,
```

Comment:
An interesting typescript error surfaces with this.. "A required parameter cannot follow an optional parameter".. the 2nd parameter, "Signer" is not optional.. and actually the first parameter "possibleRegisteredBlockNumber" isn't really optional either.. but at the call site of the "fetchSomeAnnouncement" function, the first parameter is the result of a function that fetches from state, and so can sometimes be undefined.. so rather than change the parameter order, the "| undefind" indicates the caller could pass in an undefined value.